pkg/discovery/dtofactory/pod_entity_dto_builder.go

```diff
@@ -423,7 +430,16 @@ func (builder *podEntityDTOBuilder) getPodCommoditiesBought(
 	if err != nil {
 		glog.Errorf("Failed to get %s used for current Kubernetes Cluster %s", metrics.Cluster, clusterInfo)
 	} else {
-		clusterCommodityKey, ok := clusterInfo.GetValue().(string)
+		var clusterCommodityKey string
+		var ok bool
+		if len(strings.TrimSpace(builder.clusterKeyInjected)) != 0 {
```

Comment:
This is the only place using `TrimSpace`.  We should probably make it consistent across all places.  Perhaps we do the trimming at the very top and propagate only the trimmed version down to every place?

Reply:
There is a long chain for the injected cluster key to be passed down from the command arg to pod and node build.
Even I do not assigned a value, the default empty string will be passed along. I am trying not to do the checking and trimming for empty string until this key is used for creating cluster access commodity in pod and node DTO builder.